rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindInstance.scala

```diff
@@ -8,26 +8,32 @@ import org.bitcoins.core.config.NetworkParameters
  * Created by chris on 4/29/17.
  */
 sealed trait BitcoindInstance {
-
+  require(
+    rpcUri.getPort == rpcPort,
+    s"RpcUri and the rpcPort in authCredentials are different ${rpcUri} authcred: ${rpcPort}")
   def network: NetworkParameters
   def uri: URI
   def rpcUri: URI
   def authCredentials: BitcoindAuthCredentials
+
+  def rpcPort: Int = authCredentials.rpcPort
+  def zmqPortOpt: Option[Int]
 }
 
 object BitcoindInstance {
   private case class BitcoindInstanceImpl(
     network: NetworkParameters,
     uri: URI,
     rpcUri: URI,
-    authCredentials: BitcoindAuthCredentials)
-    extends BitcoindInstance
+    authCredentials: BitcoindAuthCredentials,
+    zmqPortOpt: Option[Int]) extends BitcoindInstance
```

Comment:
Should this have default value? `None` or `Some(28332)` (28332 is used in the Bitcoin Core ZMQ doc as an example, I don't know if this is standard)

Reply:
I ended up defaulting to `None` since you need to explicitly specify the address in your `bitcoin.conf`. We also have a bad limitation here in the fact that we can only use one port, when in reality you want to have different ports for different zmq subscriber queues. 

https://github.com/bitcoin/bitcoin/blob/master/doc/zmq.md

I think we will have to address this in another PR. I've filed an issue to rectify this #262